### ISSUE TITLE
platformio misses tabulate

### ISSUE THREAD

**Issue by twam** [Homebrew/homebrew-core#153642]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew gist-logs platformio
Error: No logs.

$ brew config 
HOMEBREW_VERSION: 4.1.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7f745d04d90eaec76c8d5830d6ce6a0d7db888b3
Last commit: 2 days ago
Core tap JSON: 08 Nov 11:06 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_EDITOR: '/Applications/Sublime Text.app/Contents/SharedSupport/bin/subl' --wait
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit skylake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1-x86_64
CLT: 15.0.0.0.1.1694021235
Xcode: 15.0.1

$ brew doctor
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Compile project with `platform`.

### What happened (include all command output)?

`platformio` fails with:

```
Traceback (most recent call last):
  File "/usr/local/bin/platformio", line 5, in <module>
    from platformio.__main__ import main
  File "/usr/local/Cellar/platformio/6.1.11_2/libexec/lib/python3.12/site-packages/platformio/__main__.py", line 21, in <module>
    from platformio import __version__, exception, maintenance
  File "/usr/local/Cellar/platformio/6.1.11_2/libexec/lib/python3.12/site-packages/platformio/maintenance.py", line 29, in <module>
    from platformio.system.prune import calculate_unnecessary_system_data
  File "/usr/local/Cellar/platformio/6.1.11_2/libexec/lib/python3.12/site-packages/platformio/system/prune.py", line 19, in <module>
    from tabulate import tabulate
ModuleNotFoundError: No module named 'tabulate'
```

### What did you expect to happen?

`platformio` runs.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install platformio
platformio
```

**Comment by twam** [comment]
As `/usr/local/Cellar/python-tabulate/0.9.0/lib` only contained
```
drwxr-xr-x   4 twam  admin  128 Oct  6  2022 .
drwxr-xr-x  10 twam  admin  320 Nov  8 12:13 ..
drwxr-xr-x   3 twam  admin   96 Oct  6  2022 python3.10
drwxr-xr-x   3 twam  admin   96 Oct  6  2022 python3.11
```
I reinstalled `python-tabulate` by
```
brew uninstall platformio
brew uninstall python-tabulate  
```
and then reinstalled `platformio` by
```
brew install platformio
```
This solved the issue.

So somehow the dependency management of `python-tabulate` is broken when python version change.

**Comment by SMillerDev** [comment]
Probably https://github.com/Homebrew/brew/pull/16152

**Comment by cho-m** [comment]
In addition to https://github.com/Homebrew/brew/pull/16152, we missed a revision bump to `python-tabulate` so `python@3.12` support was not automatically installed for users.

With `brew` issue fixed, the remaining issue should hopefully be fixed with #154145
